### PR TITLE
Upgrade Cloudwatch to Slack Lambda for Node 18

### PR DESCRIPTION
**Pull request by coverbeck**
**Description**
Normally the code running on Node 14 would just continue to work, but the AWS Node 18 lambda environment includes AWS SDK v3; the AWS Node 14 has v2. So I upgraded the code to use the newer AWS SDK, as well as refactoring the code a little bit.

I also got rid of unused files that confused me when trying to update this lambda.

This will only work with the companion PR in dockstore-deploy, as that is where the runtime environment for the lambdas is specified.

**Issue**
SEAB-5759

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.

### CODE REVIEW

File: cloud-watch-to-slack-testing/deployment/index.js [hunk 1]
```diff
@@ -40,46 +40,38 @@ function getInstanceNameAndSendMsgToSlack(
...
+async function instanceName(instanceId) {
+  const client = new EC2Client();
+  const command = new DescribeInstancesCommand({
+    InstanceIds: [instanceId],
+  });
+  try {
+    const result = await client.send(command); // Throws if the instance is not found
+    const instance = result.Reservations[0].Instances; // Safe to assume, as previous line would have thrown
+    const tagInstanceNameKey = instance.Tags.find((tag) => "Name" === tag.Key);
+    return tagInstanceNameKey && tagInstanceNameKey.Value;
```
Comment (svonworl) [inline cloud-watch-to-slack-testing/deployment/index.js:68]: If it's ok for the result to be `undefined` if `tagInstanceNameKey` is `null`, you can use the optional chaining operator here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining
```
return tagInstanceNameKey?.Value;
```

File: cloud-watch-to-slack-testing/deployment/index.js [hunk 2]
```diff
@@ -40,46 +40,38 @@ function getInstanceNameAndSendMsgToSlack(
...
-      null,
+      friendlyName,
       targetInstanceId,
       processEventCallback
     );
   });
 }
 
+/**
+ * If the instanceId exists and has a tag whose key is "Name", returns that tag's value, otherwise returs null.
```
Comment (svonworl) [inline cloud-watch-to-slack-testing/deployment/index.js:55]: spellin

**Review by svonworl — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]
missing checklist item

File: cloud-watch-to-slack-testing/deployment/index.js (hunk 1 above)
Comment (coverbeck) [reply cloud-watch-to-slack-testing/deployment/index.js:68]: Nice, I had incorrectly assumed that was a TypeScript-only feature.

**Pull request merged**